pkg/types/azure/machinepool.go

```diff
@@ -30,8 +30,29 @@ type MachinePool struct {
 	//
 	// +optional
 	UltraSSDCapability string `json:"ultraSSDCapability,omitempty"`
+
+	// AcceleratedNetworking specifies whether to enable accelerated networking.
+	// Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, greatly improving its
+	// networking performance.
+	// eg. values: "accelerated", "basic"
+	//
+	// +optional
+	AcceleratedNetworking string `json:"acceleratedNetworking,omitempty"`
 }
 
+type AcceleratedNetworkingCapability string
+
+const (
+	// AccelerateNetworkingCapabilityEnabled is string representation of the AcceleratedNetworking Capability.
+	AcceleratedNetworkingCapabilityEnabled AcceleratedNetworkingCapability = "AcceleratedNetworkingEnabled"
+
+	// AccelerateNetworkingBasic is the default setting which means AcceleratedNetworking is disabled.
+	AcceleratedNetworkingBasic AcceleratedNetworkingCapability = "basic"
+
+	// AccelerateNetworkingAccelerated setting enables AcceleratedNetworking on a VM NIC.
+	AcceleratedNetworkingAccelerated AcceleratedNetworkingCapability = "accelerated"
```

Comment:
Values should be camel-case.
```suggestion
	AcceleratedNetworkingBasic AcceleratedNetworkingCapability = "Basic"

	// AccelerateNetworkingAccelerated setting enables AcceleratedNetworking on a VM NIC.
	AcceleratedNetworkingAccelerated AcceleratedNetworkingCapability = "Accelerated"
```